crates/epaint/src/text/fonts.rs

```diff
@@ -783,15 +783,10 @@ impl GalleyCache {
                 let job = Arc::new(job);
                 if allow_split_paragraphs && should_cache_each_paragraph_individually(&job) {
                     let galley = self.layout_each_paragraph_individuallly(fonts, job);
-                    let galley = Arc::new(galley);
-                    self.cache.insert(
-                        hash,
-                        CachedGalley {
-                            last_used: self.generation,
-                            galley: galley.clone(),
-                        },
-                    );
-                    galley
+                    // TODO(afishhh): This Galley cannot be added directly into the cache without taking
+                    //     extra precautions to make sure all component paragraph Galleys are not invalidated
+                    //     immediately next frame (since their `last_used` will not be updated).
+                    Arc::new(galley)
```

Comment:
This means we don't cache the full galley, and so do the paragraph-splitting and concatenation each frame… that's potentially a performance regression for large texts.

Better I think to just improve the cache

Reply:
So I implemented this, but I still seem to experience eviction of all the paragraphs in my practical application of this, although it doesn't seem to happen in a small test app. I've pushed the implementation for review (better name or approach for `layout_internal`?) and may investigate the weird cache evictions later (would blame deferred viewpoints but that wouldn't explain why it worked slightly better when re-splitting every frame).